sdks/java/managed/src/main/java/org/apache/beam/sdk/managed/Managed.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.managed;
+
+import com.google.auto.value.AutoValue;
+import java.util.Map;
+import java.util.regex.Pattern;
+import javax.annotation.Nullable;
+import org.apache.beam.sdk.schemas.transforms.SchemaTransform;
+import org.apache.beam.sdk.values.PCollectionRowTuple;
+
+public class Managed {
+  public static Read read() {
+    return new AutoValue_Managed_Read.Builder().setPattern(Read.PATTERN).build();
+  }
+
+  @AutoValue
+  public abstract static class Read extends SchemaTransform {
+    protected static final Pattern PATTERN =
```

Comment:
The idea was to dynamically discover the `SchemaTransform`s that map `read()/write()` etc. by using a pattern that maps to identifiers. The other option will be to explicitly list the set of `SchemaTransform`s.

Reply:
Well, a benefit of explicitly listing them out is to have a defined scope of what is accessible via `Managed.read()` vs `Managed.write()`

Reply:
I don't think we want to automatically pick up anything under, say, `beam:schematransform:org.apache.beam:.*_read.*` as managed. We should have an listing that serves the dual purpose of (1) mapping friendly names to urns and (2) enumerating explicitly what is supported. This listing could be a registry rather than a map here for extensibility.

Reply:
Sounds good. We can explicitly list instead of trying to do a pattern matching.